src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -464,7 +476,18 @@ protected void OnCreated(FileSystemEventArgs e)
         [SuppressMessage("Microsoft.Security", "CA2109:ReviewVisibleEventHandlers", MessageId = "0#", Justification = "Changing from protected to private would be a breaking change")]
         protected void OnDeleted(FileSystemEventArgs e)
         {
-            _onDeletedHandler?.Invoke(this, e);
+            InvokeOn(e, _onDeletedHandler);
+        }
+
+        private void InvokeOn(FileSystemEventArgs e, FileSystemEventHandler handler)
+        {
+            if (handler != null)
+            {
+                if (SynchronizingObject != null && SynchronizingObject.InvokeRequired)
```

Comment:
The same could be said for the `handler` changing from non-null to null between the check and the invocation, yeah?